test/snippet/core/detail/reflection.cpp

```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+
+#include <seqan3/core/detail/reflection.hpp>
+
+using namespace seqan3;
+
+//! [usage]
+#include <seqan3/core/detail/reflection.hpp>
+
+namespace foo
+{
+
+    template <typename ...type>
+    struct bar
+    {};
+
+} // namespace foo
+
+int main()
+{
+    std::cout << detail::get_display_name_v<foo::bar<char, double>>.string() << std::endl; // prints: foo::bar<char, double> >
```

Comment:
add `return 0;` (because it is shown within the code example and it is good style to give a return status)